test/sql/constraint.sql

```diff
@@ -0,0 +1,299 @@
+\o /dev/null
+\ir include/create_single_db.sql
+\o
+
+CREATE TABLE hyper (
+  time BIGINT NOT NULL,
+  device_id TEXT NOT NULL,
+  sensor_1 NUMERIC NULL DEFAULT 1 CHECK (sensor_1 > 10)
+);
+
+
+SELECT * FROM create_hypertable('hyper', 'time', chunk_time_interval => 10);
+
+--check and not-null constraints are inherited through regular inheritance.
+
+\set ON_ERROR_STOP 0
+INSERT INTO hyper(time, device_id,sensor_1) VALUES
+(1257987700000000000, 'dev2', 9);
+
+INSERT INTO hyper(time, device_id,sensor_1) VALUES
+(1257987700000000000, NULL, 11);
+
+\set ON_ERROR_STOP 1
+
+INSERT INTO hyper(time, device_id,sensor_1) VALUES
+(1257987700000000000, 'dev2', 11);
+
+INSERT INTO hyper(time, device_id,sensor_1) VALUES
+(1257987700000000000, 'dev2', 11);
+
+----------------------- UNIQUE CONSTRAINTS ------------------
+
+CREATE TABLE hyper_unique (
+  time BIGINT NOT NULL UNIQUE,
+  device_id TEXT NOT NULL,
+  sensor_1 NUMERIC NULL DEFAULT 1 CHECK (sensor_1 > 10)
+);
+
+
+SELECT * FROM create_hypertable('hyper_unique', 'time', chunk_time_interval => 10);
+
+INSERT INTO hyper_unique(time, device_id,sensor_1) VALUES
+(1257987700000000000, 'dev2', 11);
+
+INSERT INTO hyper_unique(time, device_id,sensor_1) VALUES
+(1257987800000000000, 'dev2', 11);
+
+
+\set ON_ERROR_STOP 0
+INSERT INTO hyper_unique(time, device_id,sensor_1) VALUES
+(1257987700000000000, 'dev2', 11);
+\set ON_ERROR_STOP 1
+
+\d+ hyper
+\d+ hyper_unique
+--should have unique constraint not just unique index
+\d+ _timescaledb_internal._hyper_2_4_chunk
+
+ALTER TABLE hyper_unique DROP CONSTRAINT hyper_unique_time_key;
+\d+ _timescaledb_internal._hyper_2_4_chunk
+
+--uniqueness not enforced
+INSERT INTO hyper_unique(time, device_id,sensor_1) VALUES
+(1257987700000000000, 'dev3', 11);
+
+
+--shouldn't be able to create constraint
+\set ON_ERROR_STOP 0
+ALTER TABLE hyper_unique ADD CONSTRAINT hyper_unique_time_key UNIQUE (time);
+\set ON_ERROR_STOP 1
+
+DELETE FROM hyper_unique WHERE device_id = 'dev3';
+
+--now can create
+ALTER TABLE hyper_unique ADD CONSTRAINT hyper_unique_time_key UNIQUE (time);
+\d+ _timescaledb_internal._hyper_2_4_chunk
+
+--test adding constraint with same name to different table -- should fail
+\set ON_ERROR_STOP 0
+ALTER TABLE hyper ADD CONSTRAINT hyper_unique_time_key UNIQUE (time);
```

Comment:
Is this actually testing anything apart from regular PostgreSQL functionality since this constraint is on the main table? Just wondering.

Reply:
I don't think so